### PR TITLE
Add ZStandard byte compressor

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

File: tests/compressor.cpp [hunk 1]
```diff
@@ -55,6 +53,17 @@ float _baseTime = 0.f;
 int main( int, char** )
 {
     registry.addDirectory( std::string( PRESSION_BUILD_DIR ) + "/lib" );
+    registry.addDirectory( "../bin" );
+    registry.addDirectory( "../lib" );
+    registry.addDirectory( "../../install/bin" );
+    registry.addDirectory( "../../install/lib" );
+    registry.addDirectory( "images" );
+    registry.addDirectory(
+        "/nfs4/bbp.epfl.ch/visualization/resources/meshes/mediumPly/" );
```
Comment (tribal-tec) [inline tests/compressor.cpp]: really? what happens if this directory does not exist?

**Comment by rdumusc** [comment]
It would be good to write a few words to motivate each of the two individual commits. On what basis where the compression parameters changed? When should one choose to use the new ZStandard over the other alternatives?

File: tests/compressor.cpp (hunk 1 above)
Comment (eile) [inline tests/compressor.cpp]: The no file is found.

**Comment by eile** [comment]
> On what basis where the compression parameters changed?

They are relative to the RLE compressor and the data compressed. Since both change over time with new compilers, CPUs and tested data, they are a somewhat arbitrary snapshot at creation time. They are determined by running the compressor benchmark which calculates them.

**Comment by eile** [comment]
> When should one choose to use the new ZStandard over the other alternatives?

There are two ways: Let Pression decide by the magic in Compressor::choose (https://github.com/Eyescale/Pression/blob/master/pression/compressor.h#L78), or pick one manually for your use case. As you can see from the speed/ratio, it's about 2x slower then the current "standard" Snappy but compresses 30% better.

**Comment by eile** [comment]
NB: I'm working on compression plugin API v2 which should be usable for normal people, so that we may start using it in Deflect and ZeroEQ eventually.

**Comment by rdumusc** [comment]
OK thanks, the important part for me is to add this information in the changelog or commit messages for future reference. Something like:
- Default parameters optimized for platform ABC using the compressor benchmark tool. This is expected to have impact XYZ on users.
- Added new ZStandard byte compressor which is about 2x slower then the current "standard" Snappy but compresses 30% better. Users can select it manually or let Compressor::choose pick it for the appropriate speed/ratio.

**Comment by eile** [comment]
Added point 1 as FAQ in README

**Comment by eile** [comment]
The profile of ZStandard is part of it's spec, imo it would be redundant to mention it again.

**Comment by eile** [comment]
Retest this please.

**Comment by rdumusc** [comment]
Well, these things are somewhat subjective, but I think it would have helped me as a reviewer to see this information in the commit messages, and by extension I though it could be useful for users and future reference. But +1 then.

**Comment by eile** [comment]
Retest this please.

**Comment by eile** [comment]
Ping - needs approval (now)

File: pression/CMakeLists.txt [hunk 2]
```diff
@@ -49,8 +51,25 @@ set(PRESSION_COMPRESSORS
...
+  compressor/zstd/lib/compress/zstd_compress.c
+  compressor/zstd/lib/decompress/huf_decompress.c
+  compressor/zstd/lib/decompress/zbuff_decompress.c
+  compressor/zstd/lib/decompress/zstd_decompress.c
+  compressor/zstd/lib/dictBuilder/divsufsort.c
+  compressor/zstd/lib/dictBuilder/zdict.c
 )
 
+include_directories(
+  ${CMAKE_CURRENT_SOURCE_DIR}/compressor/zstd/lib
```
Comment (tribal-tec) [inline pression/CMakeLists.txt]: absolute path necessary?

File: pression/compressor/compressorZSTD.cpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                               const bool /*alpha*/ )
+{
+    _nResults = 1;
+    if( _results.size() < _nResults )
+        _results.push_back( new pression::plugin::Compressor::Result );
+    size_t size = ZSTD_compressBound( nPixels );
+    _results[0]->reserve( size );
+
+    size = ZSTD_compress( _results[0]->getData(), size, inData, nPixels, 2 );
+    assert( size != 0 );
```
Comment (tribal-tec) [inline pression/compressor/compressorZSTD.cpp]: throw and assert?

**Pull request merged**